src/problems/problem_utils.jl

```diff
@@ -8,6 +8,7 @@ promote_tspan((t1,t2)::Tuple{T,S}) where {T,S} = promote(t1, t2)
 promote_tspan(tspan::Number) = (zero(tspan),tspan)
 promote_tspan(tspan::Nothing) = (nothing,nothing)
 promote_tspan(tspan::Function) = tspan
+promote_tspan(tspan::AbstractArray) = length(tspan) == 2 ? (first(tspan),last(tspan)) : throw(error("The length of tspan must be two (and preferably, tspan should be a tuple, i.e. (0.0,1.0)). If you are trying to include other values for saving reasons, note see the [common solver arguments page](https://docs.juliadiffeq.org/latest/basics/common_solver_opts/) for information on the saving command saveat."))
```

Comment:
```suggestion
promote_tspan(tspan::AbstractArray) = length(tspan) == 2 ? (promote(first(tspan),last(tspan))) : throw(error("The length of tspan must be two (and preferably, tspan should be a tuple, i.e. (0.0,1.0)). If you are trying to include other values for saving reasons, note see the [common solver arguments page](https://docs.juliadiffeq.org/latest/basics/common_solver_opts/) for information on the saving command saveat."))
```

We need this to handle
```julia
julia> Real[0, 1.0]
2-element Array{Real,1}:
 0
 1.0
```